### DISCUSSION TITLE
low energy button connects, but not able to catch when button is pressed.

### DISCUSSION THREAD

**Discussion by dejhost** [hbldh/bleak#1668] [Q&A]
Hi There. 
I am on a workstation with Linux Mint. I suceed connecting to my [low energy bluetooth button,](https://www.alibaba.com/product-detail/SCiE5-Excellent-design-BLE-5-0_1600204748399.html?spm=a2700.galleryofferlist.normal_offer.d_title.550013a0vSNyOH) but I do not manage to recognize when the button is pressed.

**The CODE**

```python
import asyncio
from bleak import BleakClient

BUTTON_PRESS_UUID = "0000fff2-0000-1000-8000-00805f9b34fb"  # Update with the correct characteristic UUID

async def button_press_callback(sender: int, data: bytes):
    print("Hello World!")  # This is the message you want to print when the button is pressed.

async def main():
    device_address = "FA:8B:4F:43:E7:6B"  # Your device address
    print("Starting Bluetooth client...")  # Debug statement

    try:
        async with BleakClient(device_address) as client:
            print(f"Connected to {device_address}")

            # Get services
            services = await client.get_services()
            print(f"Services found.")

            # Print each service and its characteristics
            for service in services:
                print(f"Service: {service.uuid}")
                for characteristic in service.characteristics:
                    print(f"  Characteristic: {characteristic.uuid}")
                    if "notify" in characteristic.properties:
                        print("    Supports notifications.")
                    else:
                        print("    Does NOT support notifications.")

            # Start listening for notifications on the button press characteristic
            await client.start_notify(BUTTON_PRESS_UUID, button_press_callback)
            print(f"Listening for button presses on {BUTTON_PRESS_UUID}...")

            # Keep the program running to listen for notifications
            try:
                while True:
                    await asyncio.sleep(1)
            except KeyboardInterrupt:
                print("Exiting...")
                await client.stop_notify(BUTTON_PRESS_UUID)

    except Exception as e:
        print(f"Error: {e}")

# Run the main function
if __name__ == "__main__":
    asyncio.run(main())

```

**The Output in the terminal:**

'Starting Bluetooth client...
```console
Connected to FA:8B:4F:43:E7:6B
/media/dejhost/RnD/RnD/Speckle/temp/bleak01.py:18: FutureWarning: This method will be removed future version, use the services property instead.
  services = await client.get_services()
Services found.
Service: 0000180a-0000-1000-8000-00805f9b34fb
  Characteristic: 00002a27-0000-1000-8000-00805f9b34fb
    Does NOT support notifications.
  Characteristic: 00002a28-0000-1000-8000-00805f9b34fb
    Does NOT support notifications.
  Characteristic: 00002a29-0000-1000-8000-00805f9b34fb
    Does NOT support notifications.
  Characteristic: 00002a24-0000-1000-8000-00805f9b34fb
    Does NOT support notifications.
  Characteristic: 00002a26-0000-1000-8000-00805f9b34fb
    Does NOT support notifications.
Service: 0000fff0-0000-1000-8000-00805f9b34fb
  Characteristic: 0000fff2-0000-1000-8000-00805f9b34fb
    Supports notifications.
  Characteristic: 0000fff1-0000-1000-8000-00805f9b34fb
    Does NOT support notifications.
Service: 00001801-0000-1000-8000-00805f9b34fb
Service: 0000fe59-0000-1000-8000-00805f9b34fb
  Characteristic: 8ec90003-f315-4f60-9fb8-838830daea50
    Does NOT support notifications.
Listening for button presses on 0000fff2-0000-1000-8000-00805f9b34fb...`
```
Here is information that is shown that my bluetooth manager picks up:
![image](https://github.com/user-attachments/assets/d980b7a6-3c85-43cf-964f-48e9920e2f48)

Your help is much appreciated!

**Comment by dlech** [comment]
Are you sure that the device is actually sending notifications? You can log Bluetooth packets to be sure (see troubleshooting page of the docs).

If the button came with an example app, you could log Bluetooth packets from that and see if there is anything else you have to do to configure the button first before it will give you notifications.

**Reply by dejhost** [reply]
Thanks for your post.

There is no dicumentation for this button.

I tried for some time now using wireshark and alike, but I could not detect anything when pressing the button.

My goal is to remotely control python scripts. So when the button is presssed, certain commands are processed. 

Any recommandations for a suitable low-energy button that works for this? What is is I need to look for in the specs to make sure that it works with bleak? Something cheap and small, battery driven. Could also be several buttons in one item.
thaknks again!

**Reply by dlech** [reply]
The main thing is to find one with some documentation or someone who has reverse engineered it so that we know what the right commands to send are.

**Reply by dejhost** [reply]
It seems ridiculus, but I searched the web for a successtory - without luck. Guess happy people don't post.
How about the [flic 2?](https://flic.io/flic2) The [protocol specs](https://github-wiki-see.page/m/50ButtonsEach/flic2-documentation/wiki/Flic-2-Protocol-Specification) seem promising to me. Would you mind taking a look?

**Reply by dlech** [reply]
The protocol looks a bit complicated.

**Comment by ukBaz** [comment]
Looking at the information for the original button on the alibaba website, it looks like it is heavily using beacon formats to share information. Have you tried exploring the button using (nRF Connect) [https://www.nordicsemi.com/Products/Development-tools/nRF-Connect-for-mobile] to see if that gives any insight?

**Reply by dejhost** [reply]
I installed nRF, but it turned out too complex for my purpose. I ended up purchasing a flic2 button. That does the trick for me: I wrote a small script that calls any other script, once I press the button.
